### PR TITLE
Feature/430 add images to project

### PR DESCRIPTION
**Pull request by macfleury-2000**
## Description

A few sentences describing the overall goals of the pull request's commits.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I did not update API Controllers, if I did, I added/changed Postman tests
-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [X] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [X] New and existing unit tests pass locally with my changes
-   [X] I updated the changelog with an end-user readable description
-   [X] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Add a file (image)
2. Add a project with multiple files
3. Open project and note that the right files are added

## Link to issue

Closes: #430

### CODE REVIEW

File: API/Controllers/ProjectController.cs [hunk 1]
```diff
@@ -363,6 +363,23 @@ public async Task<IActionResult> CreateProjectAsync([FromBody] ProjectResource p
...
+            foreach(int projectResourceImageId in projectResource.ImageIds)
+            {
+                Models.File image = await fileService.FindAsync(projectResourceImageId);
+                if(image == null)
+                {
+                    ProblemDetails problem = new ProblemDetails
+                                             {
+                                                 Title = "Image was not found.",
+                                                 Detail = "The specified image was not found while creating project.",
+                                                 Instance = "BBE862F5-0F88-478A-92EF-6864860ACEC5"
```
Comment (JVerbruggen) [inline API/Controllers/ProjectController.cs:375]: Please generate a new GUID for this problem instance

File: API/Controllers/ProjectController.cs [hunk 2]
```diff
@@ -550,6 +567,23 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
...
```
Comment (JVerbruggen) [inline API/Controllers/ProjectController.cs:579]: Please generate a new GUID for this problem instance

File: API/Controllers/ProjectController.cs [hunk 3]
```diff
@@ -550,6 +567,23 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
...
+                                             };
+                    return BadRequest(problem);
+                }
+
+                project.Images.Add(image);
+            }
+
```
Comment (JVerbruggen) [inline API/Controllers/ProjectController.cs:586]: Is there a possibility to also remove images from a project?

**Review by JVerbruggen — CHANGES_REQUESTED** [review]
Looks good! Please look at the requested changes. Nice GitHub name btw xdd

**Comment by JVerbruggen** [comment]
Also take a look at why the unit & integration tests fail

File: API/Controllers/ProjectController.cs (hunk 3 above)
Comment (macfleury-2000) [reply API/Controllers/ProjectController.cs:586]: Removing happens within the update method. When you update with less images, the other will automatically be removed

**Comment by waltersajtos** [comment]
There was something weird going on with the File class types. All other changes look good, good work!

**Review by waltersajtos — APPROVED** [review]
There was something weird going on with the File class types. All other changes look good, good work!

File: Repositories/ProjectRepository.cs [hunk 4]
```diff
@@ -159,6 +159,7 @@ public override async Task<Project> FindAsync(int id)
                                     .Where(s => s.Id == id)
```
Comment (TimSnoek123) [inline Repositories/ProjectRepository.cs:159]: More a note for later, but do we need all these images in each of these functions? Maybe we should take a look at that with performance. But maybe i'm wrong and they aren't needed idk.

**Review by TimSnoek123 — APPROVED** [review]
Prima.

**Review by JVerbruggen — APPROVED** [review]
LGTM!

**Review by JVerbruggen — APPROVED** [review]

**Review by waltersajtos — APPROVED** [review]

**Pull request merged**